packages/react-component-library/src/fragments/Masthead/Masthead.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useRef, useState } from 'react'
+import ReactCSSTransitionGroup from 'react-addons-css-transition-group'
+import classNames from 'classnames'
+
+import { Link, Searchbar, ScrollableNav } from '../../components'
+import { Logo as DefaultLogo, Search as SearchIcon } from '../../icons'
+import { NotificationPanel } from '..'
+import { UserLink } from './UserLink'
+
+export interface MastheadProps {
+  homeLink?: LinkTypes
+  LinkComponent?: any
+  Logo?: React.ComponentType
+  navItems?: NavItemTypes[]
+  NotificationsPopoverContent?: JSX.Element
+  onSearch?: (term: string) => void
+  searchPlaceholder?: string
+  title: string
+  unreadNotification?: boolean
+  user?: UserType
+}
+
+export const Masthead: React.FC<MastheadProps> = ({
```

Comment:
Might be nice to build up the Masthead using composition instead of using props to initialise it. Could have broken up the code a bit more. Notably the search and notifications. This is a really chunky component. Maybe just something for us to consider in the future.

Reply:
I broke out the search bar, navbar and notification but will have one last look through.